src/modules/main/sidebar/Sidebar.tsx

```diff
@@ -1,75 +1,41 @@
-import React from 'react';
-import { NavLink, useLocation } from 'react-router-dom';
+import React, {useState, useCallback} from 'react';
 
+import { Logo } from 'ui';
 import { Button } from '@material-ui/core';
-import DashboardIcon from '@material-ui/icons/Dashboard';
-import ProjectsIcon from '@material-ui/icons/Work';
-import TemplatesIcon from '@material-ui/icons/LibraryBooks';
 
-import { Logo } from 'ui';
+import { SidebarProps } from '.';
 
-import { SidebarProps, SidebarLink } from '.';
+import {SidebarPanel} from './sidebar-panel';
+import {SidebarLinks} from './sidebar-links';
 
 import csx from './Sidebar.scss';
 
-const LINK_HEIGHT = 80;
-
-const MARKER_HEIGHT = 30;
-
-const sidebarLinks: SidebarLink[] = [
-  { label: 'Dashboard', path: '/dashboard', icon: <DashboardIcon />, exact: true },
-  { label: 'Templates', path: '/templates', icon: <TemplatesIcon /> },
-  { label: 'Projects', path: '/projects', icon: <ProjectsIcon /> }
-];
-
-const getActiveLinkIdx = (pathname: string, links: SidebarLink[]) => {
-  return links.findIndex(({ path, exact }) => {
-    if (exact) {
-      return path === pathname;
-    }
-
-    return pathname.includes(path);
-  });
-};
-
 export const Sidebar = ({ basePath }: SidebarProps) => {
-  const { pathname } = useLocation();
+  const [open, setOpen] = useState(false);
 
-  const activeLinkIdx = getActiveLinkIdx(pathname.replace(basePath, ''), sidebarLinks);
+  const toggleOpen = () => {
+      setOpen(!open);
+    }
+    
 
+  const link = (icon, label) => {
```

Comment:
rename to render link and add here useCallback with empty deps array

Reply:
The solution has been applied